ansible_ai_connect/ai/api/model_pipelines/http/pipelines.py

```diff
@@ -78,26 +78,36 @@ def task_gen_timeout(self, task_count=1):
         return self._timeout * task_count if self._timeout else None
 
     def _setup_ssl_context(self):
-        """Let ssl.create_default_context() discover certs.
-        Following container best practices - use environment variables to help
-        Python's default SSL context find mounted certificates automatically.
-        This avoids explicit certificate path management in application code.
+        """Check for mounted service-ca certificate and configure session-level
+        SSL verification instead of global environment variables.
+        This avoids affecting other parts of the application.
         """
         if self.config.verify_ssl:
             # Check for mounted service-ca certificate (container/K8s pattern)
             service_ca = settings.SERVICE_CA_PATH
             if os.path.exists(service_ca):
-                os.environ.setdefault("REQUESTS_CA_BUNDLE", service_ca)
-                os.environ.setdefault("SSL_CERT_FILE", service_ca)
-                logger.info("Configured SSL context to use mounted service-ca certificate")
+                # Store the service CA path for session-level use
+                # instead of setting global environment variables
+                self._service_ca_path = service_ca
+                logger.info(
+                    "Found mounted service-ca certificate for HTTP pipeline SSL verification"
+                )
+            else:
+                self._service_ca_path = None
```

Comment:
Do you need to set it to `None`?

You seem to check if `self` has attribute `_service_ca_path` elsewhere.

For example [here](https://github.com/ansible/ansible-ai-connect-service/pull/1756/files#diff-b6c900f30e6213219c6ddcd3af4d9405581fee6029a05e493bdd4d9613268578R105).

Reply:
Yes, I've set to None as it directs that "no service CA path available". 

And, for:
 `if self.config.verify_ssl and hasattr(self, "_service_ca_path") and self._service_ca_path:`

I can update to:
 `if self.config.verify_ssl and self._service_ca_path:`
to remove the redundant comparison